src/loggers.jl

```diff
@@ -161,6 +161,18 @@ function getpath(logger::Logger)
     return results
 end
 
+function Serialization.serialize(s::AbstractSerializer, logger::Logger)
+    try
+        invoke(serialize, Tuple{AbstractSerializer, Any}, s, logger)
+    catch e
```

Comment:
I think you should check for the specific Task serialization error here, and then throw a custom error. If the error is something else, `rethrow` without doing anything.

Reply:
Ok, I was slightly concerned about loosing the original stacktrace

Reply:
If you pass your new error to `rethrow` it will keep the old stacktrace! :)

Reply:
TIL. Should I do that in this case?

Reply:
Answer: yes